proc/foxsi_level0_to_level1.pro

```diff
@@ -212,27 +247,55 @@ FUNCTION	FOXSI_LEVEL0_TO_LEVEL1, FILENAME, DETECTOR = DETECTOR, STOP = STOP, $
-	data_struct[i3].unrel_xy[2,*,0] = 63 - data0[i3].strips[2,*]
+	data_struct[i3].unrel_xy[0,*,pside] = strip_shift[1] - data0[i3].strips[2,*]
+	data_struct[i3].unrel_xy[1,*,pside] = strip_shift[1] - data0[i3].strips[2,*]
+	data_struct[i3].unrel_xy[2,*,pside] = strip_shift[1] - data0[i3].strips[2,*]
 	; ---------------------
 
 	; Subtract common mode for hit
 	data_struct.hit_adc_sub = fix(data_struct.hit_adc) - fix(data_struct.hit_cm)
 
 	; Determine hit position
-	; First, in detector pixel coords.
-	data_struct[i0].hit_xy_det[1] = 127 - data0[i0].hit_strip[0]
-	data_struct[i1].hit_xy_det[1] = 63 - data0[i1].hit_strip[0]
-	if i2[0] ne -1 then data_struct[i2].hit_xy_det[0] = 63 - data0[i2].hit_strip[1]
-	data_struct[i3].hit_xy_det[0] = 127 - data0[i3].hit_strip[1]
-
-	; Change hit position to payload coordinates
-	data_struct.hit_xy_pay = get_payload_coords( data_struct.hit_xy_det, detector )
+  ; First, in detector pixel coords. ; distinct calculation for Si and CdTe
+  IF cdte eq 0 THEN BEGIN
+    ; this is the way it has been done for FOXSI 1 and 2 Si detectors
+    data_struct[i0].hit_xy_det[1] = 127 - data0[i0].hit_strip[0]
+    data_struct[i1].hit_xy_det[1] = 63 - data0[i1].hit_strip[0]
+    if i2[0] ne -1 then data_struct[i2].hit_xy_det[0] = 63 - data0[i2].hit_strip[1]
+    data_struct[i3].hit_xy_det[0] = 127 - data0[i3].hit_strip[1]
+    ; note that using this, we need a reflexion in y axis to have the correct coordinates, 
+    ; but historically in the FOXSI software this is done at a later stage, so 
+    ; we keep the calculation in that state for Si detector to make sure this procedure
+    ; is compatible with previous data set and codes
+  ENDIF ELSE BEGIN
+    IF year EQ 2014 THEN BEGIN
+      ; this is the way it should be done for FOXSI2 Cdte, implemented on Nov 28 2018
+      data_struct[i0].hit_xy_det[0] = 63 - data0[i0].hit_strip[0] 
+      data_struct[i1].hit_xy_det[0] = 127 - data0[i1].hit_strip[0]  
+      data_struct[i2].hit_xy_det[1] = 127 - data0[i2].hit_strip[1]
+      data_struct[i3].hit_xy_det[1] = 63 - data0[i3].hit_strip[1]
+    ENDIF
+    IF year EQ 2018 THEN BEGIN
+      ; this is the way it should be done for FOXSI3 Cdte, implemented on Nov 28 2018
+      data_struct[i0].hit_xy_det[1] = 127 - data0[i0].hit_strip[0]
+      data_struct[i1].hit_xy_det[1] = 63 - data0[i1].hit_strip[0]
+      data_struct[i2].hit_xy_det[0] = 63 - data0[i2].hit_strip[1]
+      data_struct[i3].hit_xy_det[0] = 127 - data0[i3].hit_strip[1]
+    ENDIF
+  ENDELSE
+
+  ; Change hit position to payload coordinates: makes the distinction between Si and Cdte
+  IF CDTE EQ 0 THEN BEGIN
+    xy_det = data_struct.hit_xy_det
+;    xy_det[1,*] = 127 - xy_det[1,*] ; reverse Y axis to be in the right detector coordinates
+    data_struct.hit_xy_pay = get_payload_coords( xy_det, detector, cdte=cdte )
+  ENDIF ELSE BEGIN
+    data_struct.hit_xy_pay = get_payload_coords( data_struct.hit_xy_det, detector, cdte=cdte )
+  ENDELSE
```

Comment:
It looks like these two cases produce the same result, unless I missed something!

Reply:
looks like it is again something that should have disappeared, I will remove it!